q2_types/feature_data/_format.py

```diff
@@ -100,34 +100,52 @@ class TSVTaxonomyFormat(model.TextFileFormat):
     """
     HEADER = ['Feature ID', 'Taxon']
 
-    def sniff(self):
+    def _check_tsv_tax_format(self, n=None):
         with self.open() as fh:
             data_lines = 0
             header = None
-            while data_lines < 10:
-                line = fh.readline()
+
+            file_ = enumerate(fh) if n is None else zip(range(n), fh)
+
+            for i, line in file_:
+                i = i + 1
 
                 if line == '':
                     # EOF
                     break
-                elif line.lstrip(' ') == '\n':
-                    # Blank line
+                elif line.strip(' ') == '\n':
                     continue
                 elif line.startswith('#'):
                     # Comment line
                     continue
 
-                cells = line.rstrip('\n').split('\t')
+                cells = line.strip('\n').split('\t')
+
                 if header is None:
                     if cells[:2] != self.HEADER:
-                        return False
+                        raise ValidationError("'Feature ID' and 'Taxon' must"
+                                              " be included as headers to be"
+                                              " a valid TSV file. Please check"
+                                              " header values in your file.")
                     header = cells
                 else:
                     if len(cells) != len(header):
-                        return False
+                        raise ValidationError('Number of headers are not the '
+                                              'same as number of columns in '
+                                              'the file. \nHeader values: '
+                                              '{} \nColumn values: {} '
+                                              '\nIssue on line: {}'
+                                              .format(header, cells[:], i))
+
                     data_lines += 1
 
-            return header is not None and data_lines > 0
+            if data_lines == 0:
+                raise ValidationError("No sample records found in manifest, "
```

Comment:
This error message looks like it was copy-pasted from the manifest validation code. Can you change it to indicate that these are "feature" records (not sample). We can also drop the "where" aspect of this message.

Reply:
With respect to the "where", are you referring to the `ValidationError` message that starts on line 133? The part that tells the user the line in which the error occurred?